src/VehicleSetup/FirmwareImage.cc

```diff
@@ -238,7 +238,7 @@ bool FirmwareImage::_px4Load(const QString& imageFilename)
 
     uint32_t firmwareBoardId = (uint32_t)px4Json.value(_jsonBoardIdKey).toInt();
     uint32_t actualBoardId = _boardId == Bootloader::boardIDPX4FMUV3 ? Bootloader::boardIDPX4FMUV2 : _boardId;;
```

Comment:
This line here is used to convert an actual board id to a compatible one. Such that the != check works correctly. Then isCompatible kind of does the same thing. Could you toss this line and move the V3->V2 compatibility into your new routine which is better.

Reply:
oh, I didn't notice what that line was doing.